helm_deploy/hmpps-interventions-service/reports/crs_performance_report.sql

```diff
@@ -25,12 +31,16 @@ COPY (
     )                       AS outcomes_to_be_achieved_count,
     'coming-later'          AS outcomes_progress,
     ap.number_of_sessions   AS count_of_sessions_expected,
-    (
-      select count(app.id)
-      from action_plan_appointment app
-      where app.action_plan_id = ap.id and attended in ('YES', 'LATE')
-    )                       AS count_of_sessions_attended,
+    s.attended              AS count_of_sessions_attended,
     r.concluded_at          AS date_intervention_ended,
+    (
+      CASE
+        WHEN r.concluded_at IS NOT NULL AND eosr.id IS NULL THEN 'cancelled'
```

Comment:
i'm not sure this is the correct check. 

a referral could have been completed, but an EOSR may not yet have been created. 

in code we do this check as follows:

```
    if (!hasActionPlan)
      return CANCELLED

    if (hasAttendedNoAppointments)
      return CANCELLED

    if (hasAttendedSomeAppointments && hasSubmittedEndOfServiceReport)
      return PREMATURELY_ENDED

    if (hasAttendedAllAppointments && hasSubmittedEndOfServiceReport)
      return COMPLETED
```

Reply:
Thanks @tomsmyers!

Looking at the surrounding code, in https://github.com/ministryofjustice/hmpps-interventions-service/blob/efc79fd4934c0d783ef37bf5ed059676f7cb47a5/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralConcluder.kt#L31-L32

we set the `concluded_at` time if we "done" with the referral, right?

So I'm assuming if we check that `concluded_at` exists, that means the state around end of service report can no longer change. Is that correct?

Reply:
yes you're right - thanks!